src/js/helper.js

```diff
@@ -0,0 +1,15 @@
+function GUI_checkOperatingSystem() {
+    if (navigator.appVersion.indexOf("Win") !== -1) {
+        return "Windows";
+    } else if (navigator.appVersion.indexOf("Mac") !== -1) {
+        return "MacOS";
+    } else if (navigator.appVersion.indexOf("Android") !== -1) {
+        return "Android";
+    } else if (navigator.appVersion.indexOf("Linux") !== -1) {
+        return "Linux";
+    } else if (navigator.appVersion.indexOf("X11") !== -1) {
+        return "UNIX";
+    } else {
+        return "Unknown";
+    }
+}
```

Comment:
`navigator.appVersion` has been deprecated. We already have this function in gui.js but it is not an esm module yet.

```
function GUI_checkOperatingSystem() {
    return navigator?.userAgentData?.platform || 'Android';
}
```

Reply:
removed helper.js, the reference conflict got removed and its not needed in port_handler anymore.